client/js/components/statement/splitStatement/SideBar.vue

```diff
@@ -8,107 +8,201 @@
 </license>
 
 <template>
-  <div class="side-bar">
-    <dp-label
-      :text="Translator.trans('tags')"
-      for="searchSelect"
-      class="u-mb-0" />
-
-    <div
-      v-if="editingSegment && editingSegment.tags.length > 0"
-      class="u-mt-0_5 u-mb-0_5">
+  <div
+    ref="sideBar"
+    class="side-bar flex flex-col">
+
+    <!-- Selected Tags Section -->
+    <div class="relative px-2 pt-2">
+      <dp-label
+        :text="Translator.trans('tags')"
+        for="searchSelect"
+        class="u-mb-0" />
       <div
-        v-for="(tag, idx) in editingSegment.tags"
-        :key="`tag_${idx}`"
-        class="tag inline-flex font-size-small u-mr-0_25 u-mb-0_25">
-        {{ tag.tagName }}
-        <button
-          type="button"
-          class="tag__remove btn--blank o-link--default u-ml-0_25"
-          @click="removeTag(tag.id)">
-          <dp-icon
-            icon="close"
-            size="small" />
-        </button>
+        v-if="editingSegment && editingSegment.tags.length > 0"
+        class="flex flex-wrap gap-1 bg-white">
+        <div
+          v-for="(tag, idx) in editingSegment.tags"
+          :key="`tag_${idx}`"
+          :class="assignTagSizeClasses(tag,idx)">
+          <div
+            :class="[
+              'tag flex whitespace-nowrap overflow-hidden text-sm px-0.5 py-0.5',
+              isTagAppliedToSegment(tag.id) ? 'bg-gray-500': 'bg-green-400',
+              isLastTagWithEvenPosition(idx) ? 'w-fit' : ''
+            ]"
+          >
+          <span class="overflow-hidden text-ellipsis">
+            {{ tag.tagName }}
+          </span>
+            <button
+              type="button"
+              class="tag__remove btn--blank o-link--default ml-1"
+              data-cy="sidebar:removeTag"
+              @click="removeTag(tag.id)">
+              <dp-icon
+                icon="close"
+                size="small" />
+            </button>
+          </div>
+        </div>
       </div>
-    </div>
 
-    <div class="u-mb">
-      <!-- search available tags -->
-      <search-select
-        v-if="showCreateForm === false"
-        @open-create-form="showCreateForm = true"
-        :selected="selectedTags"
-        :place-holder="Translator.trans('tag.search')"
-        :options="searchableTags" />
-
-      <!-- create tags + topics -->
-      <dp-create-tag
-        v-else
-        @close-create-form="showCreateForm = false" />
+      <!--   Floating Context Button -->
+      <button
```

Comment:
I'd propose to make this a local component, too. Or, would there be so many different props?